src/presentation/validations/login/form-fields-type.ts

```diff
@@ -0,0 +1,48 @@
+import * as yup from 'yup'
+import { FieldPatternMap } from '../core/field-pattern-map'
+import { FieldsValidationWrapper } from '../core/form-validation-wrapper'
+
+export type LoginFormFields = {
+    user: string
+    password: string
+}
+
+export interface LoginFieldsValidationWrapper
+    extends FieldsValidationWrapper<LoginFormFields> {}
+
+export class LoginFieldsValidationWrapperImpl
+    implements LoginFieldsValidationWrapper
+{
+    onPasswordNotFulfilled: string = 'Campo obrigatório'
+
+    onUserNotFulfilled: string = 'Campo obrigatório'
+
+    onUserPatternUnmatched: string = 'O campo deve ser preenchido corretamente'
+
+    patterns: FieldPatternMap<LoginFormFields> = {
+        user: {
+            matcher: /^\d{2}\.\d{3}\.\d{3}\/\d{4}-\d{2}$/,
```

Comment:
Adicionar comentário para deixar claro q isso se trata de um regex de CNPJ

```suggestion
            // Regex for CNPJ - 00.000.000/0000-00
            matcher: /^\d{2}\.\d{3}\.\d{3}\/\d{4}-\d{2}$/,
```